src/lib.rs

```diff
@@ -455,12 +519,31 @@ impl<T: Runtime> Client<T> {
             self.account(signer.account_id(), None).await?.nonce
         };
         let call = self.encode(call)?;
+        let era_info = if let MortalPeriod::Mortal(mortal_period) =
+            self.signed_options.mortal_period
+        {
+            let current_block = match self.block(None::<T::Hash>).await? {
+                Some(signed_block) => signed_block.block,
+                None => return Err("RPC chain_getBlock returned None when Some(signed_block) was expected".into()),
+            };
+            let current_number =
+                (*current_block.header().number()).saturated_into::<u64>();
+            let current_hash = current_block.hash();
+
+            (
+                Era::mortal(mortal_period, current_number),
+                Some(current_hash),
```

Comment:
dq: why do we need to return both the block number and the block hash? Isn't that redundant info? Could we get away with only the hash and return only `Era::Mortal(period, current_hash)`/`Era::Immortal`?

Reply:
`Era::mortal` is a function that returns the enum variant `Era::Mortal`. The function takes in a mortal period and block number, does some math and spits out the variant `Mortal`, which holds the mortal period and a derived `phase` - so the `Mortal` variant is not aware of the block number or block hash. See: https://docs.rs/sp-runtime/2.0.0/src/sp_runtime/generic/era.rs.html#66-76

We need the `current_hash` in addition to the `Mortal` variant because the `additional_signed` for `CheckEra` needs to return the hash of the block. For `Era::Immortal` we just pass `None` for the block hash because we end up just using the genesis hash, which we already know.